### ISSUE TITLE
twisted.web.http.Request._sendError is untested, dead code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6122]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6122|
|Type|defect|
|Created|2012-10-16 13:13:14Z|

It's a nice idea, but none of the implementation actually uses it.  Deleting it would make `Request` shorter.  Changing all the parts of `Request` that send errors to use it might make `Request` even shorter still.  (found while porting http.py to Python 3)

Attachments:

* [6122.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d94/d942735accca32f39871e370c54c9a8ff6ad3295/103a79b4e945d3d8fb301d1d4c38f72a533cea1d.patch) (748 bytes) - added by moijes12 on 2013-04-05 08:37:36Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6122 6122
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1350393194000000 1350393194000000
changetime__1365191375000000 1365191375000000
version__None None
owner__exarkun exarkun
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)<a name="note_2"></a>|@moijes12 commented|
|-|-|

Shouldn't we first deprecate it ?

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)<a name="note_3"></a>|@moijes12 commented|
|-|-|

Also, the only places where I can think Request._sendError can be used is Request.requestReceived and maybe Request.setResponseCode. So it may be better to remove _sendError altogether. I have attached a patch with the changes. The entire regression test have been run and no unexpected errors or failures are seen.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

> Shouldn't we first deprecate it ?

No.  It isn't part of the public API, so the backwards compatibility policy does not apply to it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Thanks!  This looks pretty good.  I am going to change the news fragment from a removal to a misc, though.  Since this is a private API (as I mentioned above), this removal doesn't really need to be mentioned in the news file.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [37945]) Apply 6122.patch - remove dead helper method in twisted.web.http

Author: moijes12
Reviewer: exarkun
Fixes: [#6122](https://github.com/twisted/twisted/issues/6122)
